client/trino-cli/src/main/java/io/trino/cli/Query.java

```diff
@@ -228,7 +231,6 @@ private void renderUpdate(PrintStream out, QueryStatusInfo results)
             status += format(": %s row%s", count, (count != 1) ? "s" : "");
         }
         out.println(status);
-        discardResults();
```

Comment:
I think this should still be called but after the call to both `renderUpdate` and `renderResults`, on line 168. If results are not consumed, queries can get stuck in the `FINISHING` phase. Also, there are no other calls to `discardResults`.

Reply:
hmm ok, I will give it a look thanks

Reply:
@nineinchnick  after `renderResults` completes, we shouldn't need to do `discardResults`, should we?

Reply:
Agreed